mtp_api/apps/security/tests/test_views/test_check_views.py

```diff
@@ -0,0 +1,1231 @@
+                'security-check-reject',
+                kwargs={'pk': check.pk},
+            ),
+            {
+                'decision_reason': 'some reason',
+                'rejection_reasons': {'payment_source_linked_other_prisoners': True}
+            },
+            format='json',
+            HTTP_AUTHORIZATION=auth,
+        )
+
+        self.assertEqual(response.status_code, http_status.HTTP_400_BAD_REQUEST)
+        self.assertDictEqual(
+            response.json(),
+            {
+                'status': ['Cannot reject an accepted check.'],
+            }
+        )
+
+        check.refresh_from_db()
+
+        self.assertEqual(check.status, CHECK_STATUS.ACCEPTED)
+
+
+class CheckAutoAcceptRuleViewTestCase(APITestCase, AuthTestCaseMixin):
+    fixtures = ['initial_types.json', 'test_prisons.json', 'initial_groups.json']
+
+    def setUp(self):
+        super().setUp()
+        self.users = make_test_users(num_security_fiu_users=2)
+        self.added_by_user = self.users['security_fiu_users'][0]
+        self.auth = self.get_http_authorization_for_user(self.users['security_fiu_users'][0])
+
+        prisoner_locations = load_random_prisoner_locations(number_of_prisoners=1)
+        generate_payments(payment_batch=1)
+
+        self.prisoner_profile = generate_prisoner_profiles_from_prisoner_locations(prisoner_locations)[0]
+        self.sender_profile = generate_sender_profiles_from_payments(number_of_senders=1, reassign_dcsd=True)[0]
+        self.debit_card_sender_details = self.sender_profile.debit_card_details.first()
+
+    def test_auto_accept_rule_create(self):
+        expected_response = {
+            'prisoner_profile': self.prisoner_profile.id,
+            'debit_card_sender_details': self.debit_card_sender_details.id,
+            'states': [
+                {
+                    'active': True,
+                    'reason': 'they have amazing hair',
+                    'added_by': {
+                        'first_name': self.added_by_user.first_name,
+                        'last_name': self.added_by_user.last_name,
+                        'username': self.added_by_user.username,
+                    }
+                }
+            ]
+        }
+        response = self.client.post(
+            reverse(
+                'security-check-auto-accept-list'
+            ),
+            data={
+                'prisoner_profile': self.prisoner_profile.id,
+                'debit_card_sender_details': self.debit_card_sender_details.id,
+                'states': [
+                    {
+                        'reason': 'they have amazing hair',
+                        'added_by': self.added_by_user.id
+                    }
+                ]
+            },
+            format='json',
+            HTTP_AUTHORIZATION=self.auth,
+        )
+        self.assertEqual(response.status_code, 201)
+        actual_response = response.json()
+        self.assertIn('id', list(actual_response.keys()))
```

Comment:
just fyi, you can simply do `self.assertIn('id', actual_response)` because dicts have a `__contains__` method that checks for a key's presence

Reply:
Fair point, do you want me to make that change here?

Reply:
up to you really! i mean this works fine